timer/src/main/java/com/toggl/timer/startedit/ui/StartEditDialogFragment.kt

```diff
@@ -280,6 +296,58 @@ class StartEditDialogFragment : BottomSheetDialogFragment() {
         }
     }
 
+    private fun startEditingTimeDate(dateTimePickMode: DateTimePickMode, editableTimeEntry: EditableTimeEntry) {
+        when (dateTimePickMode) {
+            DateTimePickMode.None -> dismissEditDialog()
+            DateTimePickMode.StartTime -> startEditingTime(editableTimeEntry.startTimeOrNow())
+            DateTimePickMode.StartDate -> startEditingDate(
+                editableTimeEntry.startTimeOrNow(),
+                maxTime = editableTimeEntry.endTimeOrNow()
+            )
+            DateTimePickMode.EndTime -> startEditingTime(editableTimeEntry.endTimeOrNow()!!)
+            DateTimePickMode.EndDate -> startEditingDate(
+                editableTimeEntry.endTimeOrNow()!!,
+                minTime = editableTimeEntry.startTime!!
+            )
+        }
+    }
+
+    private fun dismissEditDialog() {
+        editDialog?.dismiss()
+        editDialog = null
+    }
+
+    private fun startEditingTime(initialTime: OffsetDateTime) {
+        val timePickerDialog = TimePickerDialog(
+            requireContext(),
+            null,
+            initialTime.hour, initialTime.minute, true
+        )
+
+        timePickerDialog.setOnCancelListener(dispatchingCancelListener)
+
+        timePickerDialog.show()
+        editDialog = timePickerDialog
+    }
+
+    private fun startEditingDate(
+        initialTime: OffsetDateTime,
+        minTime: OffsetDateTime? = null,
+        maxTime: OffsetDateTime? = null
+    ) {
+        val datePickerDialog =
+            DatePickerDialog(requireContext(), null, initialTime.year, initialTime.monthValue - 1, initialTime.dayOfMonth)
+        val datePicker = datePickerDialog.datePicker
+
+        maxTime?.let { datePicker.maxDate = maxTime.toEpochMillisecond() }
+        minTime?.let { datePicker.minDate = minTime.toEpochMillisecond() }
+
+        datePickerDialog.setOnCancelListener(dispatchingCancelListener)
+
+        datePickerDialog.show()
+        editDialog = datePickerDialog
```

Comment:
How about something like:
```suggestion
        with(DatePickerDialog(
            requireContext(),
            null,
            initialTime.year,
            initialTime.monthValue - 1,
            initialTime.dayOfMonth
        )) {
            maxTime?.let { datePicker.maxDate = maxTime.toEpochMillisecond() }
            minTime?.let { datePicker.minDate = minTime.toEpochMillisecond() }
            setOnCancelListener(dispatchingCancelListener)
            show()
            editDialog = this
        }
```

Reply:
Or maybe better:
```suggestion
        editDialog = DatePickerDialog(
            requireContext(),
            null,
            initialTime.year,
            initialTime.monthValue - 1,
            initialTime.dayOfMonth
        ).apply {
            maxTime?.let { datePicker.maxDate = maxTime.toEpochMillisecond() }
            minTime?.let { datePicker.minDate = minTime.toEpochMillisecond() }
            setOnCancelListener(dispatchingCancelListener)
            show()
        }
```

Reply:
I like the second suggestion 👍